DevExtreme.AspNet.TagHelpers.Generator/Descriptor.cs

```diff
@@ -0,0 +1,73 @@
+﻿using DevExtreme.AspNet.TagHelpers.Generator.IntellisenseData;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DevExtreme.AspNet.TagHelpers.Generator {
+
+    public class Descriptor {
+        IDictionary<string, Descriptor> _innerDescriptors;
+
+        public readonly string RawType;
+        public readonly string RawName;
+        public readonly string Summary;
+        public readonly string[] AllowedValues;
+
+        public bool IsChildTag;
+
+        public Descriptor(Info info) {
+            RawType = info.Type;
+            RawName = info.Name;
+            Summary = Utils.NormalizeDescription(info.Description);
+            IsChildTag = info.Props.Any();
+
+            if(info.AllowedValues != null) {
+                AllowedValues = info.AllowedValues
+                    .Select(i => i.Trim(' ', '\'', '"'))
+                    .Where(i => i != "undefined")
+                    .OrderBy(i => i)
+                    .ToArray();
+            } else {
+
+            }
+
+            if(info.Props.Any()) {
+                _innerDescriptors = info.Props
+                    .Select(i => new Descriptor(i))
+                    .ToDictionary(d => d.RawName, d => d, StringComparer.OrdinalIgnoreCase);
+            }
+        }
+
+        public Descriptor(Descriptor d) {
+            RawType = d.RawType;
+            RawName = d.RawName;
+            Summary = d.Summary;
+            IsChildTag = d.IsChildTag;
+            AllowedValues = d.AllowedValues;
+            _innerDescriptors = d._innerDescriptors.ToDictionary(entry => entry.Key, entry => entry.Value, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public Descriptor(Info info, string rawNameOverride)
```

Comment:
How about `new Descriptor(info) { RawName = "" }` instead of constructor overload?